### PR TITLE
Dashboard and interview detail changes

### PR DESCRIPTION
**Pull request by igtyagi**
Dashboard and interview detail changes

### CODE REVIEW

File: src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+using HT.Interview.ChatBot.Common.Contracts;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using HT.Framework.MVC;
+using HT.Interview.ChatBot.Common.Entities;
+
+namespace HT.Interview.ChatBot.API.Controllers
+{
+    [Route("api/v1/dashboard")]
+    [ApiController]
```
Comment (RavindraKhapate) [inline src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs]: Why use [ApiController], we already have mentioned in the base class.Please remove

File: src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    public class DashboardController : ApiControllerBase
```
Comment (RavindraKhapate) [inline src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs]: Please add comments

File: src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    {
+        private readonly IMapper _mapper;
+        private readonly IHttpClient _httpClient;
+        private readonly IDashboardService _dashboardDataService;
+
+        public DashboardController(IChatBotDataFactory factory)
```
Comment (RavindraKhapate) [inline src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs]: Please add comments

File: src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            _httpClient = factory.GetHttpClient();
+            _mapper = factory.GetMapperService();
+            _dashboardDataService = factory.GetDashboardService();
+        }
+
+        [HttpGet(Common.Constants.Get)]
+        public async Task<ActionResult> GetDashboardData()
```
Comment (RavindraKhapate) [inline src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs]: Please add comments.  Change method name to GetDashboardDataAsync. Everything we are using is async.

File: src/HT.Interview.ChatBot.API/Controllers/InterviewController.cs [hunk 5]
```diff
@@ -62,6 +64,12 @@ public async Task<ActionResult> GetAsync([FromQuery] QueryRequest q)
             return await GetResponseAsync(async () => await Task.FromResult(queryResponse));
         }
 
+        [HttpGet(Common.Constants.InterviewDetail)]
```
Comment (RavindraKhapate) [inline src/HT.Interview.ChatBot.API/Controllers/InterviewController.cs]: Please add comments.  Change method name to Async. Everything we are using is async.

File: src/HT.Interview.ChatBot.API/Controllers/InterviewController.cs [hunk 6]
```diff
@@ -62,6 +64,12 @@ public async Task<ActionResult> GetAsync([FromQuery] QueryRequest q)
...
+        public async Task<ActionResult> GetInterviewDetail(int userId)
```
Comment (RavindraKhapate) [inline src/HT.Interview.ChatBot.API/Controllers/InterviewController.cs]: Please add comments.  Change method name to Async. Everything we are using is async.

File: src/HT.Interview.ChatBot.API/DTO/Response/DashboardResponse.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace HT.Interview.ChatBot.API.DTO.Response
```
Comment (RavindraKhapate) [inline src/HT.Interview.ChatBot.API/DTO/Response/DashboardResponse.cs]: Add comments for the entire class

File: src/HT.Interview.ChatBot.Services/InterviewService.cs [hunk 8]
```diff
@@ -78,6 +81,26 @@ public async Task<Response> AddInterviewAsync(int userId, string dialogflowGener
             }
         }
 
+        /// <summary>
+        /// Add interview async
+        /// </summary>
+        /// <param name="userId"></param>
+        /// <param name="dialogflowGeneratedIntentId"></param>
+        /// <returns></returns>
+        public async Task<Response<IEnumerable<InterviewDetail>>> GetInterviewDetail(int userId)
```
Comment (RavindraKhapate) [inline src/HT.Interview.ChatBot.Services/InterviewService.cs]: remove try catch block

File: src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs (hunk 1 above)
Comment (igtyagi) [reply src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs]: Removed.

File: src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs (hunk 2 above)
Comment (igtyagi) [reply src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs]: Comments Added.

File: src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs (hunk 3 above)
Comment (igtyagi) [reply src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs]: Comments Added.

File: src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs (hunk 4 above)
Comment (igtyagi) [reply src/HT.Interview.ChatBot.API/Controllers/DashboardController.cs]: Comments added and method name changed.

File: src/HT.Interview.ChatBot.API/Controllers/InterviewController.cs (hunk 6 above)
Comment (igtyagi) [reply src/HT.Interview.ChatBot.API/Controllers/InterviewController.cs]: Comments added and method name changed.

File: src/HT.Interview.ChatBot.API/DTO/Response/DashboardResponse.cs (hunk 7 above)
Comment (igtyagi) [reply src/HT.Interview.ChatBot.API/DTO/Response/DashboardResponse.cs]: Comments added.

File: src/HT.Interview.ChatBot.API/appsettings.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {
```
[the remark this answers is not in the document]
Comment (igtyagi) [reply src/HT.Interview.ChatBot.API/appsettings.json]: OK, will take care of it.

File: src/HT.Interview.ChatBot.Common/Contracts/IDashboardService.cs [hunk 10]
```diff
@@ -0,0 +1,14 @@
+﻿using HT.Framework;
+using HT.Interview.ChatBot.Common.Entities;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace HT.Interview.ChatBot.Common.Contracts
+{
```
[the remark this answers is not in the document]
Comment (igtyagi) [reply src/HT.Interview.ChatBot.Common/Contracts/IDashboardService.cs]: Comments added.

File: src/HT.Interview.ChatBot.Common/Contracts/IDashboardService.cs [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+    public interface IDashboardService
+    {
+        Task<Response<IEnumerable<Dashboard>>> GetDashboardData();
```
[the remark this answers is not in the document]
Comment (igtyagi) [reply src/HT.Interview.ChatBot.Common/Contracts/IDashboardService.cs]: Comments added and method name changed.

File: src/HT.Interview.ChatBot.Common/Contracts/IInterviewService.cs [hunk 12]
```diff
@@ -8,6 +10,13 @@ namespace HT.Interview.ChatBot.Common.Contracts
     /// </summary>
     public interface IInterviewService
     {
+        /// <summary>
+        /// Get interview detail
+        /// </summary>
+        /// <param name="userId"></param>
+        /// <returns></returns>
+        Task<Response<IEnumerable<InterviewDetail>>> GetInterviewDetail(int userId);
```
[the remark this answers is not in the document]
Comment (igtyagi) [reply src/HT.Interview.ChatBot.Common/Contracts/IInterviewService.cs]: Comments added and method name changed.

File: src/HT.Interview.ChatBot.Common/Contracts/IInterviewService.cs [hunk 13]
```diff
@@ -8,6 +10,13 @@ namespace HT.Interview.ChatBot.Common.Contracts
     /// </summary>
     public interface IInterviewService
     {
+        /// <summary>
+        /// Get interview detail
+        /// </summary>
+        /// <param name="userId"></param>
+        /// <returns></returns>
```
[the remark this answers is not in the document]
Comment (igtyagi) [reply src/HT.Interview.ChatBot.Common/Contracts/IInterviewService.cs]: Comments added and method name changed.

File: src/HT.Interview.ChatBot.Common/Entities/Dashboard.cs [hunk 14]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace HT.Interview.ChatBot.Common.Entities
+{
+    public class Dashboard
```
[the remark this answers is not in the document]
Comment (igtyagi) [reply src/HT.Interview.ChatBot.Common/Entities/Dashboard.cs]: Comments added.

File: src/HT.Interview.ChatBot.Data/IChatbotDataContext.cs [hunk 15]
```diff
@@ -74,6 +74,14 @@ public interface IChatBotDataContext : IDbContext
...
-        DbSet<Model.Interview> Interview { get; set; } 
+        DbSet<Model.Interview> Interview { get; set; }
+
+        /// <inheritdoc />
+        /// <summary>
+        /// Interview entity
+        /// </summary>
+        DbSet<InterviewDetail> InterviewDetail { get; set; }
+
+        DbSet<Dashboard> DashboardData { get; set; }
```
[the remark this answers is not in the document]
Comment (igtyagi) [reply src/HT.Interview.ChatBot.Data/IChatbotDataContext.cs]: Comments Added.

File: src/HT.Interview.ChatBot.Services/ChatBotDataFactory.cs [hunk 16]
```diff
@@ -128,5 +128,10 @@ public IInterviewService GetInterviewService()
         {
             return _dependencyResolver.Resolve<IInterviewService>();
```
[the remark this answers is not in the document]
Comment (igtyagi) [reply src/HT.Interview.ChatBot.Services/ChatBotDataFactory.cs]: Comments added.

File: src/HT.Interview.ChatBot.Services/ChatBotDataFactory.cs [hunk 17]
```diff
@@ -128,5 +128,10 @@ public IInterviewService GetInterviewService()
...
         }
+
+        public IDashboardService GetDashboardService()
```
[the remark this answers is not in the document]
Comment (igtyagi) [reply src/HT.Interview.ChatBot.Services/ChatBotDataFactory.cs]: Comments added.

**Comment by igtyagi** [comment]
All changes made as per the review comments, tested and then Commit and Pushed to dev_gourav branch.

**Pull request merged**